Adding draft paper content (#55)
galaxyproject/galaxy_tool_metadata_extractor: docs/publications/BioHackrXiv/paper.md

---
title: 'How to increase the findability, visibility, and impact of Galaxy tools for your scientific community'
title_short: 'BioHackEU23 project #25: Finding Galaxy tools'
tags:
  - Findability
  - Galaxy
  - Community specific
  - Tools
  - EDAM
  - bio.tools
  - Metadata

> authors:

Reviewer: We should probably add everyone who helped during the BioHackathon as authors. What do you think?

Author: Yes, we should ask everyone that was part of the project if they would like to be added as authors.

Reviewer 2: I just asked the BH channel, please also contact everyone you think should be added.

ping:

@bgruening 
@mvdbeek

Reviewer 3: Thanks for asking but I think my contributions are to minor to be listed here.

Reviewer 4: Same, thanks for thinking of me, but I don't think I did enough to be on the paper.

Reviewer 5: Can I add myself to this preprint even if I didn't contribute that much?
I made a PR here https://github.com/supernord/galaxy_tool_extractor/pull/2
@supernord @paulzierep

Reviewer 2: @J-Swang yes please add yourself, you helped !

Reviewer 6: - [ ] @EngyNasr 
- [ ] Australian BioCommons folks

Author: Following up with the additional Australian outpost members now

Reviewer 7: I only made my contribution during the Biohackaton 2023 not to the paper, so I am not sure if that counts as contribution.

Reviewer 6: > I only made my contribution during the Biohackaton 2023 not to the paper, so I am not sure if that counts as contribution.

Of course it does!

	orcid: 0000-0003-3627-5340
	affiliation: 3
  - name: Matúš Kalaš
	orcid:
	affiliation: 4
  - name: Ove Johan Ragnar Gustafsson
	orcid: 0000-0002-2977-5032
	affiliation: 5
affiliations:
  - name: Bioinformatics Group, Department of Computer Science, University of Freiburg, Freiburg, Germany
	index: 1
  - name: Institut français de Bioinformatique, CNRS UAR 3601, Évry, France & Mésocentre Clermont-Auvergne, Université Clermont Auvergne, Aubiere, France
	index: 2
  - name: Earlham Institute, Norwich Research Park, Norwich, UK
	index: 3
  - name: 
	index: 4
  - name: Australian BioCommons, University of Melbourne, Melbourne, Victoria, Australia
	index: 5
date: 12 February 2024
bibliography: paper.bib
event: BioHackathon Europe 2023
biohackathon_name: "BioHackathon Europe 2023"
biohackathon_url:   "https://biohackathon-europe.org/"
biohackathon_location: "Barcelona, Spain, 2023"
group: BioHackrXiv
git_url: https://github.com/galaxyproject/galaxy_tool_metadata_extractor/tree/e604ffd9866f9c32f797ceee45062b4d6f2a1513/docs/publications/BioHackrXiv
authors_short: Zierep, P., Batut, B. \emph{et al.} (2024) BioHackEU23 project #25: Finding Galaxy tools
---

# Introduction or Background

> Galaxy[@10.1093/nar/gkac247] offers almost 10,000 tools that are developed across different Git repositories. Furthermore, Galaxy also embraces granular implementation of software tools as sub-modules. In practice, this means that tool suites are separated into Galaxy tools, also known as wrappers, that capture their component operations. Some key examples of suites include [Mothur](https://bio.tools/mothur)[@doi:10.1128/AEM.01541-09] and [OpenMS](https://bio.tools/openms)[@rost2016openms], which translate to tens and even hundreds of Galaxy tools. While granularity supports the sustainable development of rich domain-specific workflows, the reality is that this decentralized development and sub-module architecture makes it difficult for Galaxy users to find and reuse tools. It may also result in Galaxy developers replicating efforts by simultaneously wrapping new tools. This is further complicated by a lack of tool metadata, which prevents filtering for all tools in a specific research community or domain, and makes it all but impossible to employ advanced filtering with ontology terms and operations like EDAM[@black2021edam]. The final challenge is also an opportunity: the global nature of Galaxy means that it is a big community. Solving the visibility of tools across this ecosystem and the potential benefits are far-reaching for global collaboration on tool and workflow development.

Reviewer 8: ```suggestion
Galaxy[@10.1093/nar/gkac247] is a web-based analysis platform offering almost 10,000 different tools, which are developed across various GitHub repositories. Furthermore, Galaxy also embraces granular implementation of software tools as sub-modules. In practice, this means that tool suites are separated into Galaxy tools, also known as wrappers, that capture their component operations. Some key examples of suites include [Mothur](https://bio.tools/mothur)[@doi:10.1128/AEM.01541-09] and [OpenMS](https://bio.tools/openms)[@rost2016openms], which translate to tens and even hundreds of Galaxy tools. While granularity supports the composability of tools into rich domain-specific workflows, this decentralized development and sub-module architecture makes it difficult for Galaxy users to find and reuse tools. It may also result in Galaxy tool developers duplicating efforts by simultaneously wrapping the same software. This is further complicated by a lack of tool metadata, which prevents filtering for all tools in a specific research community or domain, and makes it all but impossible to employ advanced filtering with ontology terms and operations like EDAM[@black2021edam]. The final challenge is also an opportunity: the global nature of Galaxy means that it is a big community. Solving the visibility of tools across this ecosystem and the potential benefits are far-reaching for global collaboration on tool and workflow development.
```

---
title: 'How to increase the findability, visibility, and impact of Galaxy tools for your scientific community'

> title_short: 'BioHackEU23 project #25: Finding Galaxy tools'

Reviewer 6: ```suggestion
title_short: 'BioHackEU23 project 25: Finding Galaxy tools'
```

tags:
  - Findability
  - Galaxy

>   - Community specific

Reviewer 6: ```suggestion
  - Community-specific Galaxy tools
```

  - Tools
  - EDAM

>   - bio.tools

Author: Thoughts @nsoranzo @bebatut @paulzierep ?

>   - bio.tools

Reviewer 8: bio.tools is how it is spelled on their website, so I think we should respect that.
Same goes for microGalaxy, I believe.

>   - bio.tools

Reviewer 6: Ok then

  - Metadata
authors:
  - name: Paul Zierep*
	orcid:
	affiliation: 1
  - name: Bérénice Batut
	orcid: 0000-0001-9852-1987
	affiliation: 1, 2
  - name: Nicola Soranzo
	orcid: 0000-0003-3627-5340
	affiliation: 3
  - name: Matúš Kalaš

> 	orcid:

Reviewer 6: ```suggestion
	orcid: 0000-0002-1509-4981
```

	affiliation: 4
  - name: Ove Johan Ragnar Gustafsson
	orcid: 0000-0002-2977-5032
	affiliation: 5
affiliations:
  - name: Bioinformatics Group, Department of Computer Science, University of Freiburg, Freiburg, Germany
	index: 1

>   - name: Institut français de Bioinformatique, CNRS UAR 3601, Évry, France & Mésocentre Clermont-Auvergne, Université Clermont Auvergne, Aubiere, France

Reviewer 6: ```suggestion
  - name: Institut Français de Bioinformatique, CNRS UAR 3601, Évry, France & Mésocentre Clermont-Auvergne, Université Clermont Auvergne, Aubiere, France
```

	index: 2
  - name: Earlham Institute, Norwich Research Park, Norwich, UK
	index: 3

>   - name: 

Reviewer 6: ```suggestion
  - name: Department of Informatics, University of Bergen, Norway; and ELIXIR Norway
```

	index: 4
  - name: Australian BioCommons, University of Melbourne, Melbourne, Victoria, Australia
	index: 5

> date: 12 February 2024

Reviewer 6: ```suggestion
date: 23 February 2024
```

bibliography: paper.bib

> event: BioHackathon Europe 2023

Reviewer 6: ```suggestion
event: BH23EU
```

biohackathon_name: "BioHackathon Europe 2023"
biohackathon_url:   "https://biohackathon-europe.org/"
biohackathon_location: "Barcelona, Spain, 2023"

> group: BioHackrXiv

Reviewer 6: ```suggestion
group: Project 25 - Increasing the findability, visibility, and impact of Galaxy tools for specialised scientific Communities
```

> git_url: https://github.com/galaxyproject/galaxy_tool_metadata_extractor/tree/e604ffd9866f9c32f797ceee45062b4d6f2a1513/docs/publications/BioHackrXiv

Reviewer 6: Shouldn't this be a tree/main link instead?
And here we should consider whether we move it to the global biohackathon repo at submission time. In such case, the link should point there.
Do we know now whether it would interfere with JOSS publication or not, @bebatut ?

Reviewer 6: Another idea (radical? but software-like😊):
Use only https://github.com/galaxyproject/galaxy_tool_metadata_extractor/tree/main/docs/publications/paper.md (without the BiohackrXiv), and update it to the next paper.md (for JOSS) afterwards. (And eventual subsequent publications after that)

Author: What do you think @bebatut & @paulzierep ?

> authors_short: Zierep, P., Batut, B. \emph{et al.} (2024) BioHackEU23 project #25: Finding Galaxy tools

Reviewer 6: ```suggestion
authors_short: Paul Zierep, Bérénice Batut, \emph{et al.}
```

---

# Introduction or Background

> Galaxy[@10.1093/nar/gkac247] is a web-based analysis platform offering almost 10,000 different tools, which are developed across various GitHub repositories. Furthermore, Galaxy also embraces granular implementation of software tools as sub-modules. In practice, this means that tool suites are separated into Galaxy tools, also known as wrappers, that capture their component operations. Some key examples of suites include [Mothur](https://bio.tools/mothur)[@doi:10.1128/AEM.01541-09] and [OpenMS](https://bio.tools/openms)[@rost2016openms], which translate to tens and even hundreds of Galaxy tools. While granularity supports the composability of tools into rich domain-specific workflows, this decentralized development and sub-module architecture makes it difficult for Galaxy users to find and reuse tools. It may also result in Galaxy tool developers duplicating efforts by simultaneously wrapping the same software. This is further complicated by a lack of tool metadata, which prevents filtering for all tools in a specific research community or domain, and makes it all but impossible to employ advanced filtering with ontology terms and operations like EDAM[@black2021edam]. The final challenge is also an opportunity: the global nature of Galaxy means that it is a big community. Solving the visibility of tools across this ecosystem and the potential benefits are far-reaching for global collaboration on tool and workflow development.

Reviewer 6: ```suggestion
Galaxy[@citesAsAuthority:10.1093/nar/gkac247] is a web-based analysis platform offering almost 10,000 different tools, which are developed in various GitHub repositories. Furthermore, Galaxy also embraces granular implementation of software tools as sub-modules. In practice, this means that tool suites are separated into sets of Galaxy tools, also known as Galaxy wrappers, that contain functionality of a corresponding sub-component. Some key examples of tool suites include [Mothur](https://bio.tools/mothur)[@citesAsAuthority:doi:10.1128/AEM.01541-09] and [OpenMS](https://bio.tools/openms)[@citesAsAuthority:rost2016openms], which translate to tens and even hundreds of Galaxy tools. While granularity supports the composability of tools into diverse purpose-specific workflows, this decentralised development and modular architecture make it difficult for Galaxy users to find and use tools. It may also result in Galaxy tool-wrapper developers duplicating efforts by simultaneously wrapping the same software. This is further complicated by the lack of tool metadata, which prevents filtering of tools as relevant for a specific scientific community or domain, and makes it impossible to employ advanced filtering by ontology terms like ones from EDAM[@citesAsAuthority:black2021edam]. The final challenge is also an opportunity: the global nature of Galaxy means that it is a big community. Solving the visibility of tools across this "ecosystem", and the potential benefits are far-reaching for global collaboration on development of tools and workflows.
```

Sad the GitHub can only show a small maximum numbers of changes per line, making it unusable for editing articles in markdown 😕 In summary, I just added specifiers to the citations, and at some places changed our "Galaxish" into a more end-user English.

> To provide the research community with a comprehensive list of available Galaxy tools, a pipeline was developed at the ELIXIR BioHackathon Europe 2023 that collects Galaxy wrappers from a list of Git repositories and automatically extracts their metadata (including Conda version, bio.tools[@ison2016tools] identifiers, and EDAM annotations). The workflow also queries the availability of the tools and usage statistics from the three main Galaxy servers (usegalaxy.*). 

Reviewer 6: Can we commit this change?

Crucially, the pipeline can filter its inputs to only include tools that are relevant to a specific research community. Based on the selected filters, a community-specific interactive table is generated that can be embedded, e.g. into the respective Galaxy Hub page or Galaxy subdomain. This table allows further filtering and searching for fine-grained tool selection. The pipeline is fully automated and executes on a weekly basis. Any research community can apply the pipeline to create a table specific to their community.

> An interactive table that presents metadata is only as useful as the metadata annotations it is capturing. To improve the metadata coverage for the interactive table, the project also directly addressed the quality of tool annotations in bio.tools for the microGalaxy community: a community with a focus on tools related to microbial research. Annotation guidelines were established for this purpose, the process of updating Galaxy tool wrappers to include bio.tools identifiers was started and the outcome of these activities was evaluated using a crowdsourced approach. During the BioHackathon Europe week, the annotation practices were applied to the tools selected from the microGalaxy community. This effort allowed the team to connect more than 50 tools to their respective bio.tools entry, update the registry entry, and peer-review the result. 

Reviewer 6: ```suggestion
An interactive table that presents metadata is only as useful as the metadata annotations it is capturing. To improve the metadata coverage for the interactive table, the project also directly addressed the quality of tool annotations in bio.tools for the microGalaxy community: a community with focus on microbial research. Annotation guidelines were established for this purpose, the process of updating Galaxy tool wrappers to include bio.tools identifiers was started, and the outcome of these activities was evaluated using a crowdsourced approach. During the BioHackathon Europe week, the annotation practices were applied to the tools selected from the microGalaxy community. This effort allowed the team to connect more than 50 tools to their respective bio.tools entry, update the registry entry, and peer-review the result. 
```

Author: Not sure why this review is now `outdated`?

Author: @matuskalas

> The established pipeline and the annotation guidelines can support any research community to improve the findability, visibility, comparability, and accessibility of their Galaxy tools. Here we describe the methods and processes that resulted from this project and highlight how this will now allow the microGalaxy community to confidently navigate an ever-expanding landscape of research software in the Galaxy framework.

Reviewer 6: ```suggestion
The established pipeline and the annotation guidelines can support any scientific community to improve the findability, visibility, comparability, and accessibility of their Galaxy tools. Here we describe the methods and processes that resulted from this project, and highlight how this will now allow the microGalaxy community to confidently navigate an ever-expanding landscape of research software in the Galaxy framework.
```

# Methods

## Domain-specific interactive tools table

> Galaxy tool wrapper suites are first parsed from multiple GitHub repositories. In effect, the repositories monitored by the planemo-monitor [@Bray2022.03.13.483965] are scraped using a custom script. The planemo-monitor is part of the Galaxy tool update infrastructure and keeps track of the most up-to-date tool development repositories. 

Reviewer 6: ```suggestion
Galaxy tool-wrapper suites are first parsed from multiple GitHub repositories. In effect, the repositories monitored by the planemo-monitor [@citesAsAuthority:Bray2022.03.13.483965] are scraped using a custom script. The planemo-monitor is part of the Galaxy tool-update infrastructure, and keeps track of the most up-to-date tool development repositories. 
```

> Metadata is extracted from each tool wrapper suite. This includes: wrapper suite ID, scientific category, BioConda dependency, and a bio.tools repository URL. As a tool suite can be composed of multiple individual tools, the tool IDs for each tool are also extracted.

Reviewer 6: ```suggestion
Metadata is extracted from each tool-wrapper suite. This includes: wrapper suite ID, scientific category, Bioconda dependency, and a repository URL from bio.tools. As a tool suite can be composed of multiple individual tools, the tool IDs for each tool are also extracted.
```

> The bio.tools reference is used to request annotations via the bio.tools API, including bio.tools description, EDAM function, and EDAM operation[@black2021edam]. The conda package version is retrieved via the BioConda API and compared to the tool wrapper API to determine a tool’s update state (i.e. to update, no update required). 

Reviewer 6: ```suggestion
The bio.tools reference is used to request annotations via the bio.tools API, including bio.tools description, and functionality annotation using EDAM ontology concepts[@usesDataFrom:black2021edam]. The Conda package version is retrieved via the Bioconda API and compared to the tool wrapper API to determine a tool’s update state (_i.e._ to update, or no update required). 
```

> The Galaxy API is used to query if each tool is installed on one of the three large Galaxy servers ([usegalaxy.eu](https://usegalaxy.eu/), [usegalaxy.org](https://usegalaxy.org/), [usegalaxy.org.au](https://usegalaxy.org.au/)). Furthermore, the tool usage statistics can be retrieved from an SQL query that needs to be executed by Galaxy admins. The query used in the current implementation shows how many users executed a tool in the last 2 years on the european server ([usegalaxy.eu](https://usegalaxy.eu/)). 

Reviewer 6: ```suggestion
The Galaxy API is used to query if each tool is installed on one of the three large Galaxy servers ([usegalaxy.eu](https://usegalaxy.eu/), [usegalaxy.org](https://usegalaxy.org/), [usegalaxy.org.au](https://usegalaxy.org.au/)). Furthermore, the tool usage statistics can be retrieved from an SQL query that needs to be executed by Galaxy admins. The query used in the current implementation shows how many users executed a tool in the last 2 years on the European server ([usegalaxy.eu](https://usegalaxy.eu/)). 
```

The output of the pipeline is a table that combines Galaxy wrappers with their metadata. The complete table can then be filtered to include only tools with relevance for specific communities. The initial filtering step is based on the scientific category, which is defined for every Galaxy wrapper. These categories are high-level and cannot distinguish between specific tool functions. However, they allow for the isolation of a subgroup of the initial table for further curation. The filtered table can be manually curated by community curators. The curation step involves annotating which of the extracted tools should be kept in the final table. Curators can use the EDAM annotations and tool descriptions to assist with this curation step. The labels for each tool are stored to reduce the replication of effort even further. The practical outcome is that for repeat executions of the workflow, only new tools require curation. 

> The curated tools are transformed into an interactive web table using the data tables framework (**cite**). The table is hosted on GitHub and deployed via GitHub pages for each community. This implementation enables complex queries and filtering without the need for a database backend. The table can be embedded in any website via an iframe: examples include the Galaxy community Hub page for microGalaxy (**cite**) or a specific Galaxy subdomain (**cite**). Furthermore, a word cloud based on the usage statistics of the tools is created. 

Reviewer 6: Do we have the desired citations?

Reviewer 6: ```suggestion
The curated tools are transformed into an interactive web table using the data tables framework (**cite**). The table is hosted on GitHub and deployed via GitHub pages for each community. This implementation enables complex queries and filtering without the need for a database backend. The table can be embedded in any website via an iframe: examples include the Galaxy Community Hub page for microGalaxy (**cite**) or a specific Galaxy subdomain (**cite**). Furthermore, a word cloud based on the usage statistics of the tools is created. 
```

The workflow is scheduled via GitHub actions to run weekly, providing an up-to-date table for each community. The usage of an iframe enables updates for the table to propagate automatically to any website where it is deployed. 

> Any Galaxy community can use the pipeline by adding a folder in the [GitHub repository](https://github.com/galaxyproject/galaxy_tool_extractor/data/communities). To initialize the pipeline for a new community you need to add a list of categories to a file called categories. Additionally, tools that should be excluded or included after filtering can be added to respective files as well. A working example of the community configuration files can be found in the folder for the microgalaxy community. 

Reviewer 6: ```suggestion
Any Galaxy community can use the pipeline by adding a folder in the [GitHub repository](https://github.com/galaxyproject/galaxy_tool_extractor/data/communities). To initialise the pipeline for a new community you need to add a list of categories to a file called 'categories'. Additionally, tools that should be excluded or included after filtering, can be added to respective files as well. A working example of the community configuration files can be found in the folder for the microGalaxy community. 
```

> ![Workflow of the Galaxy tool metadata extractor pipeline. Tool wrappers are parsed from different repositories and additional metadata is retrieved from bio.tools, BioConda, and the main public Galaxy servers. Upon filtering and manual curation of the data for specific scientific communities, the data is transformed into interactive web tables and a tool usage statistic-base word cloud, that can be integrated into any website. \label{metadata_extractor_pipeline}](./figures/flowchart.png)

Reviewer 6: ```suggestion
![Workflow of the Galaxy tool metadata extractor pipeline. Tool wrappers are parsed from different repositories and additional metadata is retrieved from bio.tools, Bioconda, and the main public Galaxy servers. Upon filtering and manual curation of the data for specific scientific communities, the data is transformed into interactive web tables and a tool usage statistics-based word cloud, that can be integrated into any website. \label{metadata_extractor_pipeline}](./figures/flowchart.png)
```

## Annotation workflow

The annotation process begins by selecting a tool from a Galaxy community. This step can make use of the interactive table created by the Galaxy tool extractor scripts presented above. A curator then needs to visit the development repository of the Galaxy tool wrapper and check the *.xml file for a bio.tools xref snippet (Figure \ref{xref_snippet}). 

![xref snippet example for a Galaxy tool wrapper that contains the tool [Racon](https://bio.tools/Racon). \label{xref_snippet}](./xref.png)

bio.tools is then checked to confirm that a bio.tools identifier does, or does not, exist. The reason for this is that even if a bio.tools identifier exists in a tool wrapper, it may not necessarily exist in bio.tools. This is an observation based on real-world annotation errors and serves as a useful supporting step to improve Galaxy wrapper annotations and the completeness of the bio.tools registry. In addition, if a bio.tools identifier is not included in the wrapper, this does not mean that there is not a bio.tools identifier available in the registry.

> There are then two curation paths to choose from, depending on if a bio.tools identifier exists in the XML wrapper. In both cases, if no bio.tools entry exists, a new entry should be created and updated using the bio.tools wizard. The creation and update of an entry includes adding EDAM ontology topics and operations. This annotation process can be simplified through the use of [EDAM browser](https://edamontology.github.io/edam-browser/).

Reviewer 6: ```suggestion
There are then two curation paths to choose from, depending on whether a bio.tools identifier exists in the XML wrapper. In both cases, if no bio.tools entry exists, a new entry should be created and updated using the bio.tools wizard. The creation and update of an entry includes adding concepts from the EDAM ontology. This annotation process can be simplified through the use of [EDAM Browser](https://edamontology.github.io/edam-browser/)[@citesAsAuthority:edamBrowser].
```

In the case where no bio.tools identifier exists in the Galaxy *.xml wrapper, the development repository needs to be forked and a new branch created. A new xref snippet can then be added, and a pull request (PR) generated against the original repository.

Figure \ref{annotation_workflow} shows a step-by-step breakdown of the above process.

![Step-by-step workflow for systematically improving metadata annotations across bio.tools registry entries and Galaxy tool wrappers. After selecting a Galaxy tool and checking for the presence of a bio.tools ID in its XML file, a curator needs to review bio.tools, create a new bio.tools entry (if needed), and then ensure that both this entry and the Galaxy tool XML file are up-to-date. Updating bio.tools makes use of the registry wizard, and updating a Galaxy tool wrapper to include a bio.tools xref snippet requires a pull request against the development repository. \label{annotation_workflow}](./annotation.png)

# Outcomes and results

> There were multiple concrete outcomes from this BioHackathon project, including the interactive Galaxy tools table, a process for updating bio.tools, an in development GTN tutorial describing this process, and an update to the Galaxy IUC standards for tool wrapping. These are described in more detail below.

Reviewer 6: ```suggestion
There were multiple concrete outcomes from this BioHackathon project, including the interactive Galaxy tools table, a process for updating bio.tools, an in-development GTN tutorial describing this process, and an update to the Galaxy IUC standards for tool wrapping. These are described in more detail below.
```

## Prototype interactive table for Galaxy communities

> The described workflow for the Galaxy tool metadata extractor (see Figure \ref{metadata_extractor_pipeline}) was successfully implemented (cite GitHub) and could collect 1,294 Galaxy tool suites (cite output). Of those tool suites, only 267 had a bio.tools identifier, which highlights the importance of performing the annotation process in parallel and complementing the tools with additional metadata. An example view of the created interactive table is shown in Figure \ref{web_table}. The filtered table for the microGalaxy community is embedded in the microGalaxy Galaxy Hub (cite) page as well as the dedicated microGalaxy subdomain (cite). 

Reviewer 6: Citations? Or links?

Author: I was assuming links at this stage

> ![Screenshot of the interactive web table. The table provides comprehensive metadata for all Galaxy wrappers of a specific community and allows for custom searches based on logic filters over all columns. In the shown example, the user queries for all up-to-date tools that are annotated with the EDAM Operation assembly. \label{web_table}](./figures/itable.png)

Reviewer 6: ```suggestion
![Screenshot of the interactive web table. The table provides comprehensive metadata for all Galaxy wrappers of a specific community and allows for custom searches based on logic filters over all columns. In the shown example, the user queries for all up-to-date tools that are annotated with an EDAM operation that includes "assembly". \label{web_table}](./figures/itable.png)
```

## bio.tools and EDAM annotations for microbiome community

> During the week of the Biohackathon the microGalaxy community executed the annotation workflow as described in the Methods Section (see also Figure \ref{annotation_workflow}). The initial filtered tool table of the microGalaxy included 218 tool suites, of which 61 had corresponding bio.tools identifiers. The process of the work was tracked using a GitHub project board. After the annotation process, the number of tools with bio.tools annotations was increased to 107. The added annotations for each respective bio.tools entry were also collectively reviewed by the team. A rerun of the Galaxy tool metadata extractor pipeline collected the additional information, and the metadata is now included in the interactive microGalaxy tool table. 

Reviewer 6: ```suggestion
During the week of the Biohackathon, the microGalaxy community executed the annotation workflow as described in the Methods section (see also Figure \ref{annotation_workflow}). The initial filtered tool table of the microGalaxy included 218 tool suites, of which 61 had corresponding bio.tools identifiers. The process of the work was tracked using a GitHub project board. After the annotation process, the number of tools with bio.tools annotations was increased to 107. The added annotations for each respective bio.tools entry were also collectively reviewed by the team. A rerun of the Galaxy tool metadata extractor pipeline collected the additional information, and the metadata is now included in the interactive microGalaxy tool table. 
```

Reviewer 6: Do the numbers 61 and 107 anyhow correspond to the "> 50" mentioned higher above? I'm confused.

Author: At the start of the work 61 tools had bio.tools entries
When finished, this number had increased to 107
The number in the introduction could be update to reflect this better?

# Conclusion and outlook

The project was able to successfully meet its aim of creating reusable prototypes and processes that make the richness of the Galaxy tools ecosystem more discoverable and understandable. Central to this work was the Galaxy tool metadata extractor pipeline, which is currently generating comprehensive and interactive tabular summaries of Galaxy tools for the microbiome analysis and imaging domains within Galaxy, and which at publication can be reused by any Galaxy group or community. A set of updates to standards and processes was also created. These will support the on-going growth of the metadata hosted by the interactive tables: primarily by helping communities to maintain and extend the annotations of Galaxy tool wrappers, and the bio.tools ecosystem on which these wrapper annotations depend.

> 
> - For citations of references, we prefer the use of parenthesis, last name and year. 
> - If you use a citation manager, Elsevier – Harvard or American Psychological Association (APA) will work. If you are referencing web pages, software or so, please do so in the same way. 
> - Whenever possible, add authors and year. 
> - We have included a couple of citations along this document for you to get the idea. 
> - Please remember to always add DOI whenever available, if not possible, please provide alternative URLs. 
> - You will end up with an alphabetical order list by authors’ last name.

Reviewer 6: ```suggestion
```

# Acknowledgements

This work was developed as part of BioHackathon Europe 2023.

> 

Reviewer 6: ```suggestion
This work was supported by ELIXIR, the research infrastructure for life science data.
```

An interactive table that presents metadata is only as useful as the metadata annotations it is capturing. 
To improve the metadata coverage for the interactive table, the project also directly addressed the quality of tool annotations in bio.tools for the microGalaxy community: a community with a focus on tools related to microbial research. 
Annotation guidelines were established for this purpose, the process of updating Galaxy tool wrappers to include bio.tools identifiers was started and the outcome of these activities was evaluated using a crowdsourced approach. During the BioHackathon Europe week, the annotation practices were applied to the tools selected from the microGalaxy community. 

> This effort allowed the team to connect more than 50 tools to their respective bio.tools entry, update the registry entry, and peer-review the result. 

Reviewer 6: Is this the correct number then? (I'm still not sure I got it right😅) Thanks @supernord 🙏🏽
```suggestion
This effort allowed the team to connect additional 46 tools to their respective bio.tools entry, update the registry entry, and peer-review the result. 
```

Author: @paulzierep

![xref snippet example for a Galaxy tool wrapper that contains the tool [Racon](https://bio.tools/Racon). \label{xref_snippet}](./xref.png)

bio.tools is then checked to confirm that a bio.tools identifier does, or does not, exist. 
The reason for this is that even if a bio.tools identifier exists in a tool wrapper, it may not necessarily exist in bio.tools. 
This is an observation based on real-world annotation errors and serves as a useful supporting step to improve Galaxy wrapper annotations and the completeness of the bio.tools registry. 
In addition, if a bio.tools identifier is not included in the wrapper, this does not mean that there is not a bio.tools identifier available in the registry.

There are then two curation paths to choose from, depending on whether a bio.tools identifier exists in the XML wrapper. In both cases, if no bio.tools entry exists, a new entry should be created and updated using the bio.tools wizard. The creation and update of an entry includes adding concepts from the EDAM ontology. This annotation process can be simplified through the use of [EDAM Browser](https://edamontology.github.io/edam-browser/)[@citesAsAuthority:edamBrowser].

> In the case where no bio.tools identifier exists in the Galaxy *.xml wrapper, the development repository needs to be forked and a new branch created. 

Reviewer 8: ```suggestion
In the case where no bio.tools identifier exists in the Galaxy XML wrapper, the development repository needs to be forked and a new branch created. 
```

> A new xref snippet can then be added, and a pull request (PR) generated against the original repository.

Reviewer 8: ```suggestion
A new xref snippet can then be added, and a pull request opened against the original repository.
```

Figure \ref{annotation_workflow} shows a step-by-step breakdown of the above process.

![Step-by-step workflow for systematically improving metadata annotations across bio.tools registry entries and Galaxy tool wrappers. After selecting a Galaxy tool and checking for the presence of a bio.tools ID in its XML file, a curator needs to review bio.tools, create a new bio.tools entry (if needed), and then ensure that both this entry and the Galaxy tool XML file are up-to-date. Updating bio.tools makes use of the registry wizard, and updating a Galaxy tool wrapper to include a bio.tools xref snippet requires a pull request against the development repository. \label{annotation_workflow}](./annotation.png)

# Outcomes and results

> There were multiple concrete outcomes from this BioHackathon project, including the interactive Galaxy tools table, a process for updating bio.tools, an in development GTN tutorial describing this process, and an update to the Galaxy IUC standards for tool wrapping. 

Reviewer 8: ```suggestion
There were multiple concrete outcomes from this BioHackathon project, including the interactive Galaxy tools table, a process for updating bio.tools, an in development Galaxy Training Network (GTN) tutorial describing this process, and an update to the Galaxy IUC standards for tool wrapping. 
```

These are described in more detail below.

## Prototype interactive table for Galaxy communities

> The described workflow for the Galaxy tool metadata extractor (see Figure \ref{metadata_extractor_pipeline}) was successfully implemented ([GitHub repository](https://github.com/galaxyproject/galaxy_tool_extractor)) and could more then 1,300 Galaxy tool suites (see the [GitHub repository pages](https://galaxyproject.github.io/galaxy_tool_metadata_extractor) for an up-to-date table).

Reviewer 8: Some verb missing after `could` ?

Author: Suggest "parse" @paulzierep

Reviewer 6: ```suggestion
The described workflow for the Galaxy tool metadata extractor (see Figure \ref{metadata_extractor_pipeline}) was successfully implemented ([GitHub repository](https://github.com/galaxyproject/galaxy_tool_extractor)), and could automatically document more then 1,300 Galaxy tool suites (see the [GitHub repository pages](https://galaxyproject.github.io/galaxy_tool_metadata_extractor) for an up-to-date table).
```

Of those tool suites, only 267 had a bio.tools identifier, which highlights the importance of performing the annotation process in parallel and complementing the tools with additional metadata. 
An example view of the created interactive table is shown in Figure \ref{web_table}. 

> The filtered table for the microGalaxy community is embedded in the [Hub page for microGalaxy](https://galaxyproject.org/community/sig/microbial#tools) page as well as the dedicated [microGalaxy subdomain](https://microgalaxy.usegalaxy.eu/). 

Reviewer 8: ```suggestion
As mentioned above, the filtered table for the microGalaxy community has already been embedded in the [Hub page for microGalaxy](https://galaxyproject.org/community/sig/microbial#tools) page as well as the dedicated [microGalaxy subdomain](https://microgalaxy.usegalaxy.eu/). 
```

## bio.tools and EDAM annotations for microbiome community

During the week of the Biohackathon, the microGalaxy community executed the annotation workflow as described in the Methods section (see also Figure \ref{annotation_workflow}). The initial filtered tool table of the microGalaxy included 218 tool suites, of which 61 had corresponding bio.tools identifiers. The process of the work was tracked using a GitHub project board. After the annotation process, the number of tools with bio.tools annotations was increased to 107. The added annotations for each respective bio.tools entry were also collectively reviewed by the team. A rerun of the Galaxy tool metadata extractor pipeline collected the additional information, and the metadata is now included in the interactive microGalaxy tool table. 

## Training materials and updates to standards

> To provide the Galaxy research communities with simple and straightforward how-to to annotate their respective tool stacks, the described annotation workflow is currently being transformed into a Galaxy Training Network (GTN) tutorial. The guidelines created were also used to update the [best practices of the IUC repository](https://galaxy-iuc-standards.readthedocs.io/en/latest/best_practices/tool_xml.html).

Reviewer 8: ```suggestion
To provide the Galaxy research communities with simple and straightforward how-to to annotate their respective tool stacks, the described annotation workflow is currently being transformed into a GTN tutorial. 
The guidelines created were also used to update the [best practices for creating Galaxy tools of the IUC repository](https://galaxy-iuc-standards.readthedocs.io/en/latest/best_practices/tool_xml.html).
```

# Conclusion and outlook

The project was able to successfully meet its aim of creating reusable prototypes and processes that make the richness of the Galaxy tools ecosystem more discoverable and understandable.
Central to this work was the Galaxy tool metadata extractor pipeline, which is currently generating comprehensive and interactive tabular summaries of Galaxy tools for the microbiome analysis and imaging domains within Galaxy, and which at publication can be reused by any Galaxy group or community.
The generated tabular tool summary provides valuable information that extends beyond the use case of community tool organization, therefore an integration with the [Research Software Ecosystem (RSEc)](https://github.com/research-software-ecosystem) is currently being implemented.

> Various updates of the Galaxy tool metadata extractor pipeline are also envisioned, such as the integration of comprehensive usage statistics for all large galaxy servers, additional bio.tools meta data, and a user friendly integration of manually curation steps.

Reviewer 8: ```suggestion
Various updates of the Galaxy tool metadata extractor pipeline are also envisioned, such as the integration of comprehensive usage statistics for all large Galaxy servers, additional bio.tools meta data, and a user friendly integration of manually curation steps.
```

A set of updates to standards and processes was also created. 
These will support the on-going growth of the metadata hosted by the interactive tables: primarily by helping communities to maintain and extend the annotations of Galaxy tool wrappers, and the bio.tools ecosystem on which these wrapper annotations depend.

# Acknowledgements

This work was developed as part of BioHackathon Europe 2023.

> This work was supported by ELIXIR, the research infrastructure for life science data.

Reviewer 8: ```suggestion
This work was supported by [ELIXIR](https://elixir-europe.org), the research infrastructure for life science data.
```

While granularity supports the composability of tools into diverse purpose-specific workflows, this decentralised development and modular architecture can make it difficult for Galaxy users to find and use tools. 
It may also result in Galaxy tool-wrapper developers duplicating efforts by simultaneously wrapping the same software. 
This is further complicated by the scarcity of tool metadata, which prevents filtering of tools as relevant for a specific scientific community or domain, and makes it impossible to employ advanced filtering by ontology terms like the ones from EDAM[@citesAsAuthority:black2021edam]. 
The final challenge is also an opportunity: the global and cross-domain nature of Galaxy means that it is a big community. 
Solving the visibility of tools across this "ecosystem", and the resulting benefits, are far-reaching for the global collaborative development of data-analysis tools and workflows.

To provide the scientific community with a comprehensive list of annotated Galaxy tools, 

> we developed a pipeline at the [ELIXIR BioHackathon Europe 2023](https://2023.biohackathon-europe.org) that collects Galaxy wrappers from a list of GitHub repositories and automatically extracts their metadata (including Conda version [@citesAsAuthority:conda], bio.tools identifiers[@usesDataFrom:ison2016tools, @usesDataFrom:Ison2019], and EDAM annotations). 

Reviewer 6: ```suggestion
we developed a pipeline at the [ELIXIR BioHackathon Europe 2023](https://2023.biohackathon-europe.org), that collects Galaxy wrappers from a list of GitHub repositories and automatically extracts their metadata (including Conda version [@citesAsAuthority:conda], bio.tools identifiers[@citesAsAuthority:biotoolsSchema, @usesDataFrom:Ison2019], and EDAM annotations). 
```

> The workflow also queries the availability of the tools from the three main Galaxy servers (usegalaxy.*) as well as usage statistics from usegalaxy.eu (Note: the other servers will be queried for usage statistics in coming updates). 

Reviewer 6: ```suggestion
The workflow also queries the availability of the tools from the three main Galaxy servers (_usegalaxy.*_) as well as usage statistics from [usegalaxy.eu](https://usegalaxy.eu) (Note: the other servers will be queried for usage statistics in coming updates). 
```
Now we don't have the 3 main Galaxies listed here, so normal readers won't understand "the other" servers.

Reviewer 6: Maybe
```suggestion
The workflow also queries the availability of the tools from the three main Galaxy servers (_usegalaxy.*_) as well as usage statistics from [usegalaxy.eu](https://usegalaxy.eu) (Note: the other main Galaxy servers, [usegalaxy.org](https://usagalaxy.org) and [usegalaxy.org.au](https://usagalaxy.org.au), will be queried for usage statistics in coming updates). 
```

Crucially, the pipeline can filter its inputs to only include tools that are relevant to a specific research community. 

> Based on the selected filters, a community-specific interactive table is generated that can be embedded, e.g. into the respective [Galaxy Hub](https://galaxyproject.org/) webpage or [Galaxy subdomain](https://galaxyproject.org/eu/subdomains/). 

Reviewer 6: ```suggestion
Based on the selected filters, a community-specific interactive table is generated that can be embedded, _e.g._ into the respective [Galaxy Hub](https://galaxyproject.org/) webpage or [Galaxy subdomain](https://galaxyproject.org/eu/subdomains/). 
```

This table allows further filtering and searching for fine-grained tool selection. 
The pipeline is fully automated and executes on a weekly basis. 
Any scientific community can apply the pipeline to create a table specific to their needs.

An interactive table that presents metadata is only as useful as the metadata annotations it is capturing. 

> To improve the metadata coverage for the interactive table, the project also directly addressed the quality of tool annotations in bio.tools for the microGalaxy community: 

Reviewer 6: ```suggestion
To improve the metadata coverage for the interactive table, the project also directly addressed the quality of tool annotations in bio.tools for the [microGalaxy community](https://galaxyproject.org/community/sig/microbial/): 
```

a community with a focus on tools related to microbial research. 

> Annotation guidelines were established for this purpose, the process of updating Galaxy tool wrappers to include bio.tools identifiers was started and the outcome of these activities was evaluated using a crowdsourced approach. During the BioHackathon Europe week, the annotation practices were applied to the tools selected from the microGalaxy community. 

Reviewer 8: ```suggestion
Annotation guidelines were established for this purpose, the process of updating Galaxy tool wrappers to include bio.tools identifiers was started and the outcome of these activities was evaluated using a crowdsourced approach. During the BioHackathon Europe 2023 week, the annotation practices were applied to the tools selected from the microGalaxy community. 
```

This effort allowed the team to connect more than 50 tools to their respective bio.tools entry, update the registry entry, and collectively peer-review the results. 

The established pipeline and the annotation guidelines can support any scientific community to make their Galaxy tools more findable, visible, comparable, understandable, and accessible. 
Here, we describe the methods and processes that resulted from this project, and highlight how this will now allow the microGalaxy community to confidently navigate an ever-expanding landscape of research software in the Galaxy framework.

# Methods

## Domain-specific interactive tools table

To create the domain-specific interactive tools table, Galaxy tool-wrapper suites are first parsed from across multiple GitHub repositories. 

> In effect, the repositories monitored by the planemo-monitor [@citesAsAuthority:Bray2022.03.13.483965] are scraped using a custom script. 

Reviewer 6: ```suggestion
In effect, the repositories monitored by the planemo-monitor[@citesAsAuthority:Bray2022.03.13.483965] are scraped using a custom script. 
```

The planemo-monitor is part of the Galaxy tool-update infrastructure, and keeps track of the most up-to-date tool development repositories.

Metadata is extracted from each parsed tool-wrapper suite. 
This includes: wrapper suite ID, scientific category, Bioconda dependency, and a repository URL from bio.tools. 
As a tool suite can be composed of multiple individual tools, the tool IDs for each tool are also extracted.

> The bio.tools reference is used to request metadata annotations via the bio.tools API, including bio.tools description and functionality annotation (i.e. Operation(s)) using EDAM ontology concepts[@usesDataFrom:black2021edam]. 

Reviewer 6: ```suggestion
The bio.tools reference is used to request metadata annotations via the bio.tools API, including bio.tools description and functionality annotation using EDAM ontology concepts[@usesDataFrom:black2021edam]. 
```
I think this would be more confusing than explanatory.

If you think we need to go more into details of what kind of EDAM annotations, then there should IMHO be a separate sentence about them.

The query used in the current implementation shows the overall tool usage as well as how many users executed a tool in the last 2 years on the European server ([usegalaxy.eu](https://usegalaxy.eu/)). 

The output of the pipeline is a table that combines Galaxy wrappers with their metadata. 
The complete table can then be filtered to include only tools with relevance for specific communities. 
The initial filtering step is based on the scientific category, which is defined for every Galaxy tool wrapper. 
These categories are high-level and cannot distinguish between specific tool functions. 
However, they allow for the isolation of a subgroup of the initial table for further curation. 
The filtered table can then be manually curated by community curators. 
This curation step involves annotating which of the extracted tools should be kept in the final table. 

> Curators can use the EDAM concept annotations and tool descriptions to assist with this curation step. 

Reviewer 8: ```suggestion
Curators can use the bio.tools concept annotations and tool descriptions to assist with this curation step. 
```

I guess?

The `to keep` labels for each tool are stored to reduce the replication of effort even further. 
The practical outcome is that for repeat executions of the workflow, only new tools require curation. 

The curated tools are transformed into an interactive web table using the data tables framework[@datatables]. 
The table is hosted on GitHub and deployed via GitHub pages for each community. 
This implementation enables complex queries and filtering without the need for a database backend. 

> The table can be embedded in any website via an iframe: examples include the Galaxy community Hub page for microGalaxy [Hub page for microGalaxy](https://galaxyproject.org/community/sig/microbial#tools) or the [microGalaxy subdomain](https://microgalaxy.usegalaxy.eu/). 

Reviewer 6: ```suggestion
The table can be embedded in any website via an iframe: examples include the Galaxy community [Hub page for microGalaxy](https://galaxyproject.org/community/sig/microbial#tools) or the [microGalaxy subdomain](https://microgalaxy.usegalaxy.eu/). 
```

Furthermore, a word cloud based on the usage statistics of the tools is created. 

The workflow is run weekly via GitHub Actions continuous integration, providing an up-to-date table for each community. 
The usage of an iframe enables updates for the table to propagate automatically to any website where it is deployed. 

Any Galaxy community can use the pipeline by adding a folder in the [project GitHub repository](https://github.com/galaxyproject/galaxy_tool_extractor/). 
To initialise the pipeline for a new community you need to add a new subfolder of `data/communities/` , and inside it add a file called 'categories' with a list of Galaxy ToolShed (**cite** doi:10.1186/gb4161 ) categories. 
Additionally, tools that should be excluded or included after filtering, can be added to respective files as well. 

> A working example of the community configuration files can be found in the folder for the microgalaxy community.

Reviewer 6: ```suggestion
A working example of the community configuration files can be found in the folder for the microGalaxy community.
```

## Annotation workflow

The annotation process begins by selecting a tool from a Galaxy community. 
This step can make use of the interactive table created by the Galaxy tool extractor scripts presented above. 
A curator then needs to visit the development repository of the Galaxy tool wrapper and search the XML file for a bio.tools xref snippet (Figure \ref{xref_snippet}). 

![xref snippet example for a Galaxy tool wrapper that contains the tool [Racon](https://bio.tools/Racon). \label{xref_snippet}](./xref.png)

bio.tools is then checked to confirm that a bio.tools identifier does, or does not, exist. 
The reason for this is that even if a bio.tools identifier exists in a tool wrapper, it may not necessarily exist in bio.tools. 
This is an observation based on real-world annotation errors and serves as a useful supporting step to improve Galaxy wrapper annotations and the completeness of the bio.tools registry. 
In addition, if a bio.tools identifier is not included in the wrapper, this does not mean that there is not a bio.tools identifier available in the registry.

> There are then two curation paths to choose from, depending on whether a bio.tools identifier exists in the XML wrapper. In both cases, if no bio.tools entry exists, a new entry should be created and updated using the bio.tools wizard. The creation and update of an entry includes adding concepts from the EDAM ontology. This annotation process can be simplified through the use of [EDAM Browser](https://edamontology.github.io/edam-browser/)[@citesAsAuthority:edamBrowser].

Reviewer 6: ```suggestion
There are then two curation paths to choose from, depending on whether a bio.tools identifier exists in the XML wrapper. In both cases, if no bio.tools entry exists, a new entry should be created and updated using the bio.tools wizard. The creation and update of an entry includes adding concepts from the EDAM ontology. This annotation process can be simplified through the use of [EDAM Browser](https://edamontology.github.io/edam-browser/)[@citesAsAuthority:edamBrowser, @citesAsAuthority:edamBrowserCode].
```
I don't insist we need this additional citation of the code, and I'm pesimistic it's counted anywhere, but I'm rather curious how it will look when trying to follow the "most proper" guidelines for citing scientific software 😅

In the case where no bio.tools identifier exists in the Galaxy XML wrapper, the development repository needs to be forked and a new branch created. 
A new xref snippet can then be added, and a pull request opened against the original repository.

Figure \ref{annotation_workflow} shows a step-by-step breakdown of the above process.

![Step-by-step workflow for systematically improving metadata annotations across bio.tools registry entries and Galaxy tool wrappers. After selecting a Galaxy tool and checking for the presence of a bio.tools ID in its XML file, a curator needs to review bio.tools, create a new bio.tools entry (if needed), and then ensure that both this entry and the Galaxy tool XML file are up-to-date. Updating bio.tools makes use of the registry wizard, and updating a Galaxy tool wrapper to include a bio.tools xref snippet requires a pull request against the development repository. \label{annotation_workflow}](./annotation.png)

# Outcomes and results

> There were multiple concrete outcomes from this BioHackathon project, including the ability to create interactive Galaxy tools tables as needed for scientific communities, a process for updating bio.tools, in-development Galaxy Training Network (GTN)[@batut_community-driven_2018] tutorials describing this process, and an update to the [Galaxy IUC](https://galaxyproject.org/iuc/) tool wrapping [standards](https://galaxy-iuc-standards.readthedocs.io/en/latest/index.html). 

Reviewer 6: ```suggestion
There were multiple concrete outcomes from this BioHackathon project, including the ability to create interactive Galaxy tools tables as needed for scientific communities, a process for updating bio.tools, in-development Galaxy Training Network (GTN) tutorials[@citesAsAuthority:batut_community-driven_2018] describing this process, and an update to the [Galaxy IUC](https://galaxyproject.org/iuc/) tool wrapping [standards](https://galaxy-iuc-standards.readthedocs.io/en/latest/index.html). 
```

These are described in more detail below.

## Prototype interactive table for Galaxy communities

The described workflow for the Galaxy tool metadata extractor (see Figure \ref{metadata_extractor_pipeline}) was successfully implemented ([GitHub repository](https://github.com/galaxyproject/galaxy_tool_extractor)) and could more then 1,300 Galaxy tool suites (see the [GitHub repository pages](https://galaxyproject.github.io/galaxy_tool_metadata_extractor) for an up-to-date table).
Of those tool suites, only 267 had a bio.tools identifier, which highlights the importance of performing the annotation process in parallel and complementing the tools with additional metadata. 
An example view of the created interactive table is shown in Figure \ref{web_table}. 

> As mentioned above, the filtered table for the microGalaxy community has already been embedded in the [Hub page for microGalaxy](https://galaxyproject.org/community/sig/microbial#tools) page as well as the dedicated [microGalaxy subdomain](https://microgalaxy.usegalaxy.eu/). 

Reviewer 6: ```suggestion
As mentioned above, the filtered table for the microGalaxy community has already been embedded in the [Hub page for microGalaxy](https://galaxyproject.org/community/sig/microbial#tools), as well as the dedicated [microGalaxy subdomain](https://microgalaxy.usegalaxy.eu/). 
```

The process for creating a new interactive table for a community is currently being transformed into a GTN tutorial.

![Screenshot of the interactive web table. The table provides comprehensive metadata for all Galaxy wrappers of a specific community and allows for custom searches based on logic filters over all columns. In the shown example, the user queries for all up-to-date tools that are annotated with an EDAM operation that includes "assembly". \label{web_table}](./figures/itable.png)

> ## bio.tools and EDAM annotations for microbiome community

Reviewer 6: ```suggestion
## bio.tools and EDAM annotations for microGalaxy community
```

During the week of the BioHackathon, the microGalaxy community executed the annotation workflow as described in the Methods section (see also Figure \ref{annotation_workflow}). 
The initial filtered tool table of the microGalaxy included 218 tool suites, of which 61 had corresponding bio.tools identifiers. 
The progress of the work was tracked using a GitHub project board. 
After the annotation process, the number of tools with bio.tools annotations was increased to 107. 
The added annotations for each respective bio.tools entry were also collectively reviewed by the team. 
A rerun of the Galaxy tool metadata extractor pipeline collected the additional information, and the metadata is now included in the interactive microGalaxy tool table. 

## Training materials and updates to standards

To provide the Galaxy research communities with simple and straightforward guide to annotating their respective tool stacks, the described annotation workflow is currently being transformed into a GTN tutorial. 

> The guidelines created were also used to update the [best practices of the IUC repository](https://galaxy-iuc-standards.readthedocs.io/en/latest/best_practices/tool_xml.html).

Reviewer 8: ```suggestion
The guidelines created were also used to update the [best practices for creating Galaxy tools of the IUC repository](https://galaxy-iuc-standards.readthedocs.io/en/latest/best_practices/tool_xml.html).
```

> To provide the Galaxy research communities with simple and straightforward guide to annotating their respective tool stacks, the described annotation workflow is currently being transformed into a GTN tutorial. 

Reviewer: The tutorial is ready to be merged: https://github.com/galaxyproject/training-material/pull/4734

Reviewer: There is also a tutorial for community Galaxy tool table creation ready: https://github.com/galaxyproject/training-material/pull/4740

# Conclusion and outlook

The project was able to successfully meet its aim of creating reusable prototypes and processes that make the richness of the Galaxy tools ecosystem more discoverable and understandable.

> Central to this work was the Galaxy tool metadata extractor pipeline, which is currently generating comprehensive and interactive tabular summaries of Galaxy tools for the microbiome analysis and imaging domains within Galaxy, and which at publication can be reused by any Galaxy group or community.

Reviewer 6: ```suggestion
Central to this work was the Galaxy tool metadata extractor pipeline, which is currently generating comprehensive and interactive tabular summaries of Galaxy tools for the [microbial data](https://galaxyproject.org/community/sig/microbial/) and [image analysis](https://galaxyproject.org/use/imaging/) communities within Galaxy. The metadata extractor can be reused by any Galaxy group or community. For example, the [biodiversity and ecology](https://galaxyproject.org/use/ecology/) community with employ this pipeline in the near future[@citesAsAuthority:elixirBiodiversity].
```

> The generated tabular tool summary provides valuable information that extends beyond the use case of community tool organization, therefore an integration with the [Research Software Ecosystem (RSEc)](https://github.com/research-software-ecosystem) is currently being explored.

Reviewer 6: ```suggestion
The generated tabular tool summary provides valuable information that extends beyond the use case of listing community tools. Therefore, an integration with the [Research Software Ecosystem (RSEc)](https://github.com/research-software-ecosystem/content)[@citeAsAuthority:RSEc] is currently being worked on.
```

> Various updates of the Galaxy tool metadata extractor pipeline are also envisioned, such as the integration of comprehensive usage statistics for all large Galaxy servers, additional bio.tools meta data, and a user friendly integration of manually curation steps.

Reviewer 6: ```suggestion
Various updates of the Galaxy tool metadata extractor pipeline are also envisioned, such as the integration of comprehensive usage statistics for all large Galaxy servers, additional bio.tools metadata, and a user-friendly integration of manual curation steps.
```
